inductiva/structures/scenarios/plate_linear_elastic/geometry_utils.py

```diff
@@ -0,0 +1,614 @@
+
+import gmsh
+import json
+import math
+
+
+class Plate(ABC):
+    """Abstract base class for plates."""
+
+    @abstractmethod
+    def perimeter(self):
+        """Abstract method to calculate the perimeter of the plate."""
+        pass
+
+    @abstractmethod
+    def to_dict(self):
+        """Abstract method to convert the plate properties to a dictionary."""
+        pass
+
+    @abstractmethod
+    def to_occ(self):
+        """Abstract method to convert the plate to OpenCASCADE CAD."""
+        pass
+
+    @abstractmethod
+    def get_boundary_ids(self):
+        """Abstract method to get the IDs of the plate boundaries."""
+        pass
+
+    @abstractmethod
+    def get_mesh_metrics(self):
+        """Abstract method to get the metrics for mesh generation in Gmsh."""
+        pass
+
+
+class RectangularPlate(Plate):
+    """Rectangular plate.
+
+    Attributes:
+        plate_type (str): Plate type.
+        length (float): Plate length.
+        width (float): Plate width.
+    """
+
+    def __init__(self, length: float, width: float) -> None:
+        """Initializes a RectangularPlate object."""
+        self.plate_type = "rectangular"
+        self.length = length
+        self.width = width
+
+    def perimeter(self) -> float:
+        """Calculate the perimeter of the plate.
+
+        Returns:
+            float: The calculated perimeter.
+        """
+        return 2 * (self.width + self.length)
+
+    def to_dict(self) -> dict:
+        """Convert plate properties to a dictionary.
+
+        Returns:
+            dict: Plate properties.
+        """
+        return {
+            "plate_type": self.plate_type,
+            "length": self.length,
+            "width": self.width
+        }
+
+    def to_occ(self) -> int:
+        """Converts plate to OpenCASCADE CAD representation.
+
+        Returns:
+          plate_gmsh (int): The Gmsh entity ID representing the plate's 
+            OpenCASCADE CAD representation.
+        """
+        plate_gmsh = gmsh.model.occ.addRectangle(x=0,
+                                                 y=0,
+                                                 z=0,
+                                                 dx=self.width,
+                                                 dy=self.length)
+
+        return plate_gmsh
+
+    def get_boundary_ids(self, plate_gmsh: int) -> List[int]:
+        """Gets the IDs of the plate boundaries.
+
+        Args:
+            plate_gmsh (int): The Gmsh entity ID representing the plate's 
+              OpenCASCADE CAD representation.
+
+        Returns:
+            List[int]: A list of boundary IDs corresponding to the plate's 
+              boundaries.
+        """
+        plate_boundaries = gmsh.model.getBoundary([(2, plate_gmsh)])
+
+        return [[boundary_id[1]][0] for boundary_id in plate_boundaries]
+
+    def get_mesh_metrics(self) -> Tuple[float, float]:
+        """Gets the metrics for mesh generation in Gmsh.
```

Comment:
The naming here is a bit odd, since you're not getting metrics of the "mesh" itself, but rather of the plate. You can leave it as is for now, and we can revisit this in the future.

Reply:
i will change the name of the method to "get_plate_mesh_params" to make it clear that the method is providing parameters for meshing the plate in Gmsh